src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,199 @@
+            {
+                var isPattern = (IsPatternExpressionSyntax)propertyPattern.Parent;
+                var typeSymbol = semanticModel.GetTypeInfo(isPattern.Expression, cancellationToken).Type;
+                if (typeSymbol != null)
+                {
+                    return typeSymbol;
+                }
+            }
+
+            // switch (e) { case { $$
+            // switch (e) { case { P1: 0, $$
+            if (propertyPattern.IsParentKind(SyntaxKind.CasePatternSwitchLabel))
+            {
+                var casePattern = (CasePatternSwitchLabelSyntax)propertyPattern.Parent;
+                var switchStatment = casePattern.Parent?.Parent as SwitchStatementSyntax;
+                if (switchStatment != null && switchStatment.Expression != null)
+                {
+                    var typeSymbol = semanticModel.GetTypeInfo(switchStatment.Expression).Type;
+                    if (typeSymbol != null)
+                    {
+                        return typeSymbol;
+                    }
+                }
+            }
+
+            // e is { P1: { $$
+            // e is { P1: { P2: 2, $$
+            if (propertyPattern.IsParentKind(SyntaxKind.SubpatternElement))
+            {
+                var containingProperty = ((SubpatternElementSyntax)propertyPattern.Parent).NameColon?.Name;
+                if (containingProperty != null)
+                {
+                    var symbol = semanticModel.GetSymbolInfo(containingProperty).Symbol;
+
+                    switch (symbol)
+                    {
+                        case null:
+                            break;
+                        case IPropertySymbol propertySymbol:
+                            return propertySymbol.Type;
+                        case IFieldSymbol fieldSymbol:
+                            return fieldSymbol.Type;
+                    }
+                }
+            }
+
+            // PROTOTYPE(NullableReferenceTypes): 
+            // All the cases above should be replaced by using the semantic model's GetTypeInfo (ConvertedType) once that is integrated
+
+            return default;
+        }
+
+        // List the members that are already tested in this property sub-pattern
+        private static HashSet<string> GetTestedMembers(SyntaxTree tree, int position, CancellationToken cancellationToken)
+        {
+            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken)
+                .GetPreviousTokenIfTouchingWord(position);
+
+            // We should have gotten back a { or ,
+            if (token.IsKind(SyntaxKind.CommaToken, SyntaxKind.OpenBraceToken))
+            {
+                if (token.Parent is PropertySubpatternSyntax subpattern)
+                {
+                    return new HashSet<string>(subpattern.SubPatterns.Select(
+                        p => p.NameColon?.Name?.Identifier.ValueText).Where(s => !string.IsNullOrEmpty(s)));
```

Comment:
if you have subpatterns, can you not have a NameColon or Name?  Are those really optional?  if so... i'm curious why....

Reply:
See [error syntax](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRIMaOQSwG4SoAOsMECAdsgDQwhTIzUAmIA1AD4ACADAARcAjAG4AsACguAZgEAmPgGE+Abwl91AmVwAsfALIAKAJQq1G8wH0+AXj4wAFjmR8nKviD6CAvmPHmvEl5AA=), which shows no `NameColon`.
I don't know about no `Name`, but I preferred to be defensive.